tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts

```diff
@@ -0,0 +1,504 @@
+        }.bind(this)
+      )
+      .attr(
+        'fill',
+        function(this: AnnotationComponent, d: string) {
+          return `rgb(${this.rgbColors[0]})`;
+        }.bind(this)
+      );
+
+    indicators.exit().remove();
+  }
+
+  private drawRunHintTexts() {
+    const hintTexts = this.runHintGroup
+      .selectAll<SVGTextElement, unknown>('.hint-text')
+      .data([...this.runs]);
+
+    const hintTextEnters = hintTexts
+      .enter()
+      .append('text')
+      .attr('x', 25)
+      .attr('font-size', '10px')
+      .attr('alignment-baseline', 'middle')
+      .attr('clip-path', 'url(#hint-clip)');
+
+    hintTextEnters
+      .merge(hintTexts)
+      .attr(
+        'y',
+        function(this: AnnotationComponent, d: string) {
+          return this.yScale(d)! + 15;
+        }.bind(this)
+      )
+      .attr('class', `hint-text ${this.textClass}`)
+      .text((d: string) => d);
+
+    hintTexts.exit().remove();
+  }
+
+  private drawBars() {
+    const bars = this.barsGroup
+      .selectAll<SVGRectElement, unknown>('.bar')
+      .data(this.data);
+
+    const barEnters = bars
+      .enter()
+      .append('rect')
+      .attr('class', 'bar')
+      .attr('height', 20);
+
+    barEnters
+      .merge(bars)
+      .attr('fill', (d: ValueData) => {
+        if (d.nPMIValue === null) {
+          return '';
+        } else if (d.nPMIValue >= 0) {
+          return d3.interpolateBlues(d.nPMIValue);
+        } else {
+          return d3.interpolateReds(d.nPMIValue * -1);
+        }
+      })
+      .attr(
+        'x',
+        function(this: AnnotationComponent, d: ValueData) {
+          return this.xScale(d.metric)!;
+        }.bind(this)
+      )
+      .attr(
+        'y',
+        function(this: AnnotationComponent, d: ValueData) {
+          return this.yScale(d.run)! + 5;
+        }.bind(this)
+      )
+      .attr(
+        'width',
+        function(this: AnnotationComponent, d: ValueData) {
+          if (d.nPMIValue === null) {
+            return 0;
+          } else {
+            return this.sizeScale(Math.abs(d.nPMIValue));
+          }
+        }.bind(this)
+      );
+
+    bars.exit().remove();
+  }
+
+  private drawCountDots() {
+    const countDots = this.countDotsGroup
+      .selectAll<SVGCircleElement, unknown>('.count-dot')
+      .data(this.data);
+
+    const countDotEnters = countDots
+      .enter()
+      .append('circle')
+      .attr('class', 'count-dot')
+      .attr('stroke', 'black');
+
+    countDotEnters
+      .merge(countDots)
+      .attr(
+        'fill',
+        function(this: AnnotationComponent, d: ValueData) {
+          if (d.countValue === null) {
+            return '';
+          } else {
+            return d3.interpolateGreys(d.countValue / this.maxCount);
+          }
+        }.bind(this)
+      )
+      .attr(
+        'cx',
+        function(this: AnnotationComponent, d: ValueData) {
+          return this.xScale(d.metric)! + 70;
```

Comment:
I still think it is a good idea to write count label based on variables by factoring out values like this.